### ISSUE TITLE
update() - ColumnNotFoundException when join used

### ISSUE THREAD

**Issue by mrcnpdlk** [skipperbent/pecee-pixie#93]
For queries like this pseudocode:
```php
$qb
  ->table('foo')
  ->leftJoin('bar')
  ->where('bar.id',1)
  ->update(['foo.status'=>1])
;
```
`ColumnNotFoundException` is thrown:
```
Unknown column 'bar.id' in 'where clause'
```

For `delete()` method also problem is appeared, but in my opinion `delete()` method should have argument `table` to know which table have to be deleted;

**Comment by skipperbent** [comment]
Hi @mrcnpdlk.

Some of the logic in the `BaseAdapter` from the `select` method needs to be moved into a more generic method, so the correct query can be built for both when using `update` and `delete` statements.

I'll start work in this right away, thanks for pointing out the issue 👍 

/ Simon

**Comment by skipperbent** [comment]
Fixed in Version 4.10.0

**Comment by mrcnpdlk** [comment]
Is better but not good.
Everything works fine when we DON'T use the alias.
For example:
```php
    $res = $oQuery
        ->table(['tbl_users' => 'u'])
        ->leftJoin(['tbl_eyes', 'e'], 'u._eyeId', '=', 'e.id')
        ->where('u.age', '=', 5)
        ->update(['u.name' => 'Jan'])
    ;
```
generate query like this:
```
UPDATE `tbl_users` LEFT JOIN `tbl_eyes` AS `e` ON `u`.`_eyeId` = `e`.`id` SET `u`.`name`='Jan' WHERE `u`.`age` = 5
```
should be:
```
UPDATE `tbl_users` as u LEFT JOIN `tbl_eyes` AS `e` ON `u`.`_eyeId` = `e`.`id` SET `u`.`name`='Jan' WHERE `u`.`age` = 5
```

The alias for main table is missing.
Similar problem is for `delete()` method
